system-int-test/test/cmr/system_int_test/ingest/generics_test.clj

```diff
@@ -251,10 +252,132 @@
     ;;The second delete will not say the draft is already deleted because it no longer exists in the database
     (is (= result1 {:concept-id concept-id, :revision-id 2,:warnings nil, :existing-errors nil}))
     (is (= result2 {:errors ["CollectionDraft with native id [NativeId] in provider [PROV1] does not exist."]}))
-
+    
     ;;Verify that searching for the concept from elastic search won't return anything after the delete.
     (let [result (search-generics-test/search-request "collection-drafts" (str "concept_id=" concept-id))
           status (:status result)
           body (:body result)]
       (is (= status 200))
       (is (string/includes? body "<hits>0</hits>")))))
+
+;; Test that collection-draft can be published
+;; and the draft is removed from metadatadb.
+(deftest test-collection-draft-publishing
+  ;; Drafts have permissions to ingest on PROV1, but not on PROV2.
+  (let [;; Ingest a collection draft
+        cd-native-id "CD-NativeId"
+        coll-native-id "COLL-NativeId"
+        coll-draft (gen-util/ingest-generic-document
+                    nil "PROV1" cd-native-id :collection-draft gen-util/collection-draft :post)
+        cd-concept-id (:concept-id coll-draft)
+        coll-published (ingest/publish-non-variable-draft
+                        cd-concept-id coll-native-id nil)
+        coll-concept-id (:concept-id coll-published)
+
+        _ (index/wait-until-indexed)
+        ;; search for the published collection from elastic search should return result.
+        search-coll-result (search-generics-test/search-request "collections" (str "concept_id=" coll-concept-id))
+        search-coll-status (:status search-coll-result)
+        search-coll-body (:body search-coll-result)
+
+        ;; search for the draft from elastic search should return nothing.
+        search-draft-result (search-generics-test/search-request "collection-drafts" (str "concept_id=" cd-concept-id))
+        search-draft-status (:status search-draft-result)
+        search-draft-body (:body search-draft-result)
+       
+        ;; deleting the draft should fail. 
+        draft-delete-result (gen-util/ingest-generic-document
+                             nil "PROV1" cd-native-id :collection-draft gen-util/collection-draft :delete)]
+    ;;Verify that searching for the published collection from elastic search does return result.
```

Comment:
super nit; the comments have a space after the ";" in some of the comments but, in others it does not I think in the styling guide we put a space on each one

Reply:
Done